sdk/eventhub/azure-eventhubs/azure/eventhub/eventprocessor/_ownership_manager.py

```diff
@@ -0,0 +1,135 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# -----------------------------------------------------------------------------------
+
+import time
+import random
+import math
+from collections import Counter
+from azure.eventhub.aio import EventHubClient
+
+
+class OwnershipManager(object):
+    """Increases or decreases the number of partitions owned by an EventProcessor
+    so the number of owned partitions are balanced among multiple EventProcessors
+
+    An EventProcessor calls claim_ownership() of this class every x seconds,
+    where x is set by keyword argument "polling_interval" in EventProcessor,
+    to claim the ownership of partitions, create tasks for the claimed ownership, and cancel tasks that no longer belong
+    to the claimed ownership.
+
+    """
+    def __init__(self, event_processor, eventhub_client: EventHubClient, ownership_timeout: int):
+        self.all_parition_ids = []
+        self.eventhub_client = eventhub_client
+        self.eventhub_name = eventhub_client.eh_name
+        self.consumer_group_name = event_processor._consumer_group_name
+        self.owner_id = event_processor._id
+        self.partition_manager = event_processor._partition_manager
+        self.ownership_timeout = ownership_timeout
+
+    async def claim_ownership(self):
+        """Claims ownership for this EventProcessor
+        1. Retrieves all partition ids of an event hub from azure event hub service
+        2. Retrieves current ownership list via this EventProcessor's PartitionManager.
+        3. Balances number of ownership. Refer to _balance_ownership() for details.
+        4. Claims the ownership for the balanced number of partitions.
+
+        :return: List[Dict[Any]]
+        """
+        if not self.all_parition_ids:
+            await self._retrieve_partition_ids()
+        to_claim = await self._balance_ownership()
+        claimed_list = await self._claim_ownership(to_claim)
```

Comment:
Having the same name (except for leading underscore), but with slightly different functionality is somewhat confusing. The `claim_ownership` method actually updates (or rebalances) owerships. `_claim_ownership` actually does the claim...

Reply:
I removed _claim_ownership since it's a very short function. 
It can be implemented with one line of code.